charts/springboot/templates/deployment.yaml

```diff
@@ -58,6 +58,19 @@ spec:
                     app: {{ template "name" . }}
                     release: {{ .Release.Name | quote }}
       {{- end }}
+      {{- if .Values.topologySpreadConstraints }}
+      topologySpreadConstraints:
+{{ toYaml .Values.topologySpreadConstraints | indent 8 }}
```

Comment:
You should be able to use `nindent` instead of `indent` to avoid this "weird" indentation:
```
topologySpreadConstraints: {{ toYaml .Values.topologySpreadConstraints | nindent 8 }}
```

Reply:
Changed